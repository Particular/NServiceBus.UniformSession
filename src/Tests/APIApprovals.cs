namespace NServiceBus.UniformSession.Tests
{
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;
    using UniformSession;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void ApproveNServiceBus()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(IUniformSession).Assembly, excludeAttributes: new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" });
            Approver.Verify(publicApi);
        }
    }
}