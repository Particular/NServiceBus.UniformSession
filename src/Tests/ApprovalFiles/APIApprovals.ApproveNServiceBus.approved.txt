[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace NServiceBus.UniformSession
{
    public interface IUniformSession
    {
        System.Threading.Tasks.Task Publish(object message, NServiceBus.PublishOptions options);
        System.Threading.Tasks.Task Publish<T>(System.Action<T> messageConstructor, NServiceBus.PublishOptions publishOptions);
        System.Threading.Tasks.Task Send(object message, NServiceBus.SendOptions options);
        System.Threading.Tasks.Task Send<T>(System.Action<T> messageConstructor, NServiceBus.SendOptions options);
    }
    public class static IUniformSessionExtensions
    {
        public static System.Threading.Tasks.Task Publish(this NServiceBus.UniformSession.IUniformSession session, object message) { }
        public static System.Threading.Tasks.Task Publish<T>(this NServiceBus.UniformSession.IUniformSession session) { }
        public static System.Threading.Tasks.Task Publish<T>(this NServiceBus.UniformSession.IUniformSession session, System.Action<T> messageConstructor) { }
        public static System.Threading.Tasks.Task Send(this NServiceBus.UniformSession.IUniformSession session, object message) { }
        public static System.Threading.Tasks.Task Send<T>(this NServiceBus.UniformSession.IUniformSession session, System.Action<T> messageConstructor) { }
        public static System.Threading.Tasks.Task Send(this NServiceBus.UniformSession.IUniformSession session, string destination, object message) { }
        public static System.Threading.Tasks.Task Send<T>(this NServiceBus.UniformSession.IUniformSession session, string destination, System.Action<T> messageConstructor) { }
        public static System.Threading.Tasks.Task SendLocal(this NServiceBus.UniformSession.IUniformSession session, object message) { }
        public static System.Threading.Tasks.Task SendLocal<T>(this NServiceBus.UniformSession.IUniformSession session, System.Action<T> messageConstructor) { }
    }
}
namespace NServiceBus
{
    public class static UniformSessionExtensions
    {
        public static void EnableUniformSession(this NServiceBus.EndpointConfiguration endpointConfiguration) { }
    }
}